src/main/java/org/firmata4j/firmata/FirmataDevice.java

```diff
@@ -528,6 +496,7 @@ public void run() {
                     }
                 } catch (InterruptedException ex) {
                     LOGGER.info("FirmataParser has stopped");
+                    Thread.currentThread().interrupt();
```

Comment:
I think at this point there is no reason to interrupt the current thread because of `return` statement at the following line that finishes the thread normally.

Reply:
@kurbatov It was due to my laziness and just trying to get read of sonar-lint warning 😄 
but putting that aside, it is always recommended to make sure that we are not just clearing the interruption flag and let if go. return will just work fine. in case of intentional interrupt (this case), that should be acceptable. a better way would be to put while in the try catch and just the the interruption end the Thread. Also updating to messaging system (recommended earlier) maybe make this part also a bit more clear.

Reply:
maybe this will cover a bit of my laziness: https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html